core/server/data/migrations/versions/4.45/2022-04-19-12-23-backfill-subscriptions-offers.js

```diff
@@ -0,0 +1,77 @@
+const logging = require('@tryghost/logging');
+const DatabaseInfo = require('@tryghost/database-info');
+const {chunk} = require('lodash');
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Backfilling "offer_id" column in "members_stripe_customers_subscriptions" by matching tier and cadence');
+
+        const subquery = `
+            SELECT
+                members_stripe_customers_subscriptions.id as subscription_id,
+                offer_redemptions.offer_id as offer_id
+            FROM
+                members_stripe_customers_subscriptions
+                JOIN offer_redemptions ON offer_redemptions.subscription_id = members_stripe_customers_subscriptions.id
+                JOIN offers ON offers.id = offer_redemptions.offer_id
+                JOIN stripe_prices ON members_stripe_customers_subscriptions.stripe_price_id = stripe_prices.stripe_price_id
+                JOIN stripe_products ON stripe_prices.stripe_product_id = stripe_products.stripe_product_id
+            WHERE
+                offers.product_id = stripe_products.product_id
+                AND offers.interval = stripe_prices.interval
+                AND members_stripe_customers_subscriptions.offer_id is null
+        `;
+
+        if (DatabaseInfo.isSQLite(knex)) {
+            // Less optimized for SQLite
+            const result = await knex.raw(subquery);
+            const updatedRows = result.length;
+            const subscriptionsToUpdate = result;
+
+            // Group by offer ID (we make the assumption that we will have a limited amount of offers)
+            const groupedByOffers = {};
+            // eslint-disable-next-line no-restricted-syntax
+            for (const u of subscriptionsToUpdate) {
```

Comment:
Honestly I would simplify further for SQLite
```js
for (const u of subscriptionsToUpdate) {
  await knex('members_stripe_customers_subscriptions')
    .update('offer_id', u.offer_id)
    .where('id', u.subscription_id);
}
```

Reply:
Okay, I'll revert it to that (had this initially but was not sure about performance) 😄